some (mostly) client/gui mappings

moehreag opened the pull request.

moehreag committed:
largely gui mappings

moehreag committed:
more gui mappings

moehreag committed:
more mostly gui mappings

moehreag committed:
fix compile errors/misnames

IotaBread commented on mappings/net/minecraft/client/gui/Element.mapping:
```diff
@@ -137,5 +137,5 @@ CLASS net/minecraft/unmapped/C_fxiqpxaf net/minecraft/client/gui/Element
 			COMMENT the Y coordinate of the mouse
-		ARG 5 amount
-			COMMENT value is {@code < 0} if scrolled down, {@code > 0} if scrolled up
+		ARG 5 amountX
+		ARG 7 amountY
```
I'd add the comment back, and include it for the y amount

IotaBread commented on mappings/net/minecraft/client/gui/RenderList.mapping:
```diff
@@ -0,0 +1,17 @@
+	FIELD f_afqpfbiz layers Ljava/util/List;
+	FIELD f_lrobklrs LAYER_OFFSET F
+	METHOD m_aolowtyu (Ljava/util/function/BooleanSupplier;Lnet/minecraft/unmapped/C_nihbhmxv;Lnet/minecraft/unmapped/C_sedilmty;F)V
+		ARG 3 delta
+	METHOD m_fyitalyz renderLayers (Lnet/minecraft/unmapped/C_sedilmty;F)V
+		ARG 2 delta
```
```suggestion
		ARG 3 tickDelta
	METHOD m_fyitalyz renderLayers (Lnet/minecraft/unmapped/C_sedilmty;F)V
		ARG 2 tickDelta
```

IotaBread commented on mappings/net/minecraft/client/gui/RenderList.mapping:
```diff
@@ -0,0 +1,17 @@
+	METHOD m_moqgrgyg add (Lnet/minecraft/unmapped/C_nihbhmxv$C_lkrghfli;)Lnet/minecraft/unmapped/C_nihbhmxv;
+		ARG 1 layer
+	METHOD m_odhxuxlp render (Lnet/minecraft/unmapped/C_sedilmty;F)V
+		ARG 2 delta
```
```suggestion
		ARG 2 tickDelta
```

IotaBread commented on mappings/net/minecraft/client/gui/RenderList.mapping:
```diff
@@ -0,0 +1,17 @@
+		ARG 2 renderCondition
+	CLASS C_lkrghfli Layer
+		METHOD render render (Lnet/minecraft/unmapped/C_sedilmty;F)V
+			ARG 2 delta
```
```suggestion
			ARG 2 tickDelta
```

IotaBread commented on mappings/net/minecraft/client/gui/hud/chat/MessageStorage.mapping:
```diff
@@ -0,0 +1,73 @@
+		ARG 1 size
+	METHOD add add (ILjava/lang/Object;)V
+		ARG 1 index
+		ARG 2 object
+	METHOD addFirst addFirst (Ljava/lang/Object;)V
+		ARG 1 object
+	METHOD addLast addLast (Ljava/lang/Object;)V
+		ARG 1 object
```
I feel like these `object`s could be a bit more specific

moehreag replied:
would you prefer `element`? That's what `java.util.AbstractList` uses.

IotaBread replied:
That sounds good

IotaBread commented on mappings/net/minecraft/client/gui/hud/in_game/InGameHud.mapping:
```diff
@@ -62,7 +96,12 @@ CLASS net/minecraft/unmapped/C_nepwsmgw net/minecraft/client/gui/hud/in_game/InG
 	METHOD m_bkadrkcb shouldShowChatDisabledScreen ()Z
 	METHOD m_btqrbwce renderCrosshair (Lnet/minecraft/unmapped/C_sedilmty;F)V
 		ARG 1 graphics
+		ARG 2 delta
 	METHOD m_ciysbohv getSpectatorHud ()Lnet/minecraft/unmapped/C_ybztcyiy;
+	METHOD m_ckjzczem (Lnet/minecraft/unmapped/C_lfemghur;)Z
+		ARG 0 entry
+	METHOD m_cwobkefs renderOverlayMessage (Lnet/minecraft/unmapped/C_sedilmty;F)V
+		ARG 2 delta
```
```suggestion
		ARG 2 tickDelta
	METHOD m_ciysbohv getSpectatorHud ()Lnet/minecraft/unmapped/C_ybztcyiy;
	METHOD m_ckjzczem (Lnet/minecraft/unmapped/C_lfemghur;)Z
		ARG 0 entry
	METHOD m_cwobkefs renderOverlayMessage (Lnet/minecraft/unmapped/C_sedilmty;F)V
		ARG 2 tickDelta
```

IotaBread commented on mappings/net/minecraft/client/gui/hud/in_game/InGameHud.mapping:
```diff
@@ -77,15 +116,23 @@ CLASS net/minecraft/unmapped/C_nepwsmgw net/minecraft/client/gui/hud/in_game/InG
 		ARG 2 tickDelta
 	METHOD m_fmuufkwf renderDemoTimer (Lnet/minecraft/unmapped/C_sedilmty;F)V
 		ARG 1 graphics
+		ARG 2 delta
```
```suggestion
		ARG 2 tickDelta
```

IotaBread commented on mappings/net/minecraft/client/gui/hud/in_game/InGameHud.mapping:
```diff
@@ -77,15 +116,23 @@ CLASS net/minecraft/unmapped/C_nepwsmgw net/minecraft/client/gui/hud/in_game/InG
 	METHOD m_fydpkivs setSubtitle (Lnet/minecraft/unmapped/C_rdaqiwdt;)V
 		ARG 1 subtitle
+	METHOD m_gbnvpsmk renderSleep (Lnet/minecraft/unmapped/C_sedilmty;F)V
+		ARG 2 delta
 	METHOD m_gyqjeoze getPlayerListHud ()Lnet/minecraft/unmapped/C_lybmfibg;
+	METHOD m_hmpeaweq renderChat (Lnet/minecraft/unmapped/C_sedilmty;F)V
+		ARG 2 delta
```
```suggestion
		ARG 2 tickDelta
	METHOD m_gyqjeoze getPlayerListHud ()Lnet/minecraft/unmapped/C_lybmfibg;
	METHOD m_hmpeaweq renderChat (Lnet/minecraft/unmapped/C_sedilmty;F)V
		ARG 2 tickDelta
```

IotaBread commented on mappings/net/minecraft/client/gui/hud/in_game/InGameHud.mapping:
```diff
@@ -77,15 +116,23 @@ CLASS net/minecraft/unmapped/C_nepwsmgw net/minecraft/client/gui/hud/in_game/InG
 	METHOD m_iimewzba getBossBarHud ()Lnet/minecraft/unmapped/C_vpkfuxkl;
+	METHOD m_irzrqyso getDebugHud ()Lnet/minecraft/unmapped/C_epxplegp;
+	METHOD m_iucytvnr renderScoreboard (Lnet/minecraft/unmapped/C_sedilmty;F)V
+		ARG 2 delta
```
```suggestion
		ARG 2 tickDelta
```

IotaBread commented on mappings/net/minecraft/client/gui/hud/in_game/InGameHud.mapping:
```diff
@@ -95,13 +142,19 @@ CLASS net/minecraft/unmapped/C_nepwsmgw net/minecraft/client/gui/hud/in_game/InG
 		ARG 5 color
 	METHOD m_koznghsj renderAutosaveIndicator (Lnet/minecraft/unmapped/C_sedilmty;F)V
 		ARG 1 graphics
+		ARG 2 delta
```
```suggestion
		ARG 2 tickDelta
```

IotaBread commented on mappings/net/minecraft/client/gui/hud/in_game/InGameHud.mapping:
```diff
@@ -95,13 +142,19 @@ CLASS net/minecraft/unmapped/C_nepwsmgw net/minecraft/client/gui/hud/in_game/InG
 		ARG 2 remainTicks
 		ARG 3 fadeOutTicks
+	METHOD m_lcsxraft renderPlayerList (Lnet/minecraft/unmapped/C_sedilmty;F)V
+		ARG 2 delta
```
```suggestion
		ARG 2 tickDelta
```

IotaBread commented on mappings/net/minecraft/client/gui/hud/in_game/InGameHud.mapping:
```diff
@@ -95,13 +142,19 @@ CLASS net/minecraft/unmapped/C_nepwsmgw net/minecraft/client/gui/hud/in_game/InG
 	METHOD m_nfjwrptj getTextRenderer ()Lnet/minecraft/unmapped/C_mavozmpp;
 	METHOD m_oafggdwi renderStatusEffectOverlay (Lnet/minecraft/unmapped/C_sedilmty;F)V
 		ARG 1 graphics
+		ARG 2 delta
```
```suggestion
		ARG 2 tickDelta
```

IotaBread commented on mappings/net/minecraft/client/gui/hud/in_game/InGameHud.mapping:
```diff
@@ -120,20 +173,31 @@ CLASS net/minecraft/unmapped/C_nepwsmgw net/minecraft/client/gui/hud/in_game/InG
 	METHOD m_qblpluah updateVignetteDarkness (Lnet/minecraft/unmapped/C_astfners;)V
 		ARG 1 entity
+	METHOD m_qgtzabmx renderTitles (Lnet/minecraft/unmapped/C_sedilmty;F)V
+		ARG 2 delta
+	METHOD m_ronzofwm renderOverlays (Lnet/minecraft/unmapped/C_sedilmty;F)V
+		ARG 2 delta
```
```suggestion
		ARG 2 tickDelta
	METHOD m_ronzofwm renderOverlays (Lnet/minecraft/unmapped/C_sedilmty;F)V
		ARG 2 tickDelta
```

IotaBread commented on mappings/net/minecraft/client/gui/hud/in_game/InGameHud.mapping:
```diff
@@ -120,20 +173,31 @@ CLASS net/minecraft/unmapped/C_nepwsmgw net/minecraft/client/gui/hud/in_game/InG
 		COMMENT Renders the armor, health, air, and hunger bars.
 		ARG 1 graphics
+	METHOD m_srkhzgsd renderHotbar (Lnet/minecraft/unmapped/C_sedilmty;F)V
+		ARG 2 delta
```
```suggestion
		ARG 2 tickDelta
```

IotaBread commented on mappings/net/minecraft/client/gui/hud/in_game/InGameHud.mapping:
```diff
@@ -120,20 +173,31 @@ CLASS net/minecraft/unmapped/C_nepwsmgw net/minecraft/client/gui/hud/in_game/InG
+		ARG 0 size
 	METHOD m_twidsnlh getRiddenEntity ()Lnet/minecraft/unmapped/C_usxaxydn;
+	METHOD m_unxvdhni (Lnet/minecraft/unmapped/C_sedilmty;F)V
+		ARG 2 delta
```
```suggestion
		ARG 2 tickDelta
```

IotaBread commented on mappings/net/minecraft/client/gui/hud/in_game/InGameHud.mapping:
```diff
@@ -144,11 +208,17 @@ CLASS net/minecraft/unmapped/C_nepwsmgw net/minecraft/client/gui/hud/in_game/InG
 	METHOD m_wjrmovwk tickPausable ()V
 	METHOD m_wkdzfhws setDefaultTitleFade ()V
+	METHOD m_wscbnfkt (Lnet/minecraft/unmapped/C_sedilmty;F)V
+		ARG 2 delta
+	METHOD m_wttpxmtb (Lnet/minecraft/unmapped/C_sedilmty;F)V
+		ARG 2 delta
```
```suggestion
		ARG 2 tickDelta
	METHOD m_wttpxmtb (Lnet/minecraft/unmapped/C_sedilmty;F)V
		ARG 2 tickDelta
```

IotaBread commented on mappings/net/minecraft/client/gui/hud/in_game/InGameHud.mapping:
```diff
@@ -144,11 +208,17 @@ CLASS net/minecraft/unmapped/C_nepwsmgw net/minecraft/client/gui/hud/in_game/InG
 	METHOD m_xqvrdrrt getHeartRows (I)I
 		ARG 1 heartCount
+	METHOD m_yahflzho renderSurvivalHotbar (Lnet/minecraft/unmapped/C_sedilmty;F)V
+		ARG 2 delta
```
```suggestion
		ARG 2 tickDelta
```

IotaBread commented on mappings/net/minecraft/client/gui/hud/in_game/InGameHud.mapping:
```diff
@@ -161,5 +231,18 @@ CLASS net/minecraft/unmapped/C_nepwsmgw net/minecraft/client/gui/hud/in_game/InG
+		FIELD f_rphpsjvq hardcoreFullBlinking Lnet/minecraft/unmapped/C_ncpywfca;
+		FIELD f_slrltipg hardcoreFull Lnet/minecraft/unmapped/C_ncpywfca;
+		FIELD f_wcxevweg fullBlinking Lnet/minecraft/unmapped/C_ncpywfca;
+		METHOD <init> (Ljava/lang/String;ILnet/minecraft/unmapped/C_ncpywfca;Lnet/minecraft/unmapped/C_ncpywfca;Lnet/minecraft/unmapped/C_ncpywfca;Lnet/minecraft/unmapped/C_ncpywfca;Lnet/minecraft/unmapped/C_ncpywfca;Lnet/minecraft/unmapped/C_ncpywfca;Lnet/minecraft/unmapped/C_ncpywfca;Lnet/minecraft/unmapped/C_ncpywfca;)V
```
Empty method

moehreag replied:
I'm not entirely sure, do you mean the constructor? It's not empty, it just only assigns the fields to the parameters passed.

IotaBread replied:
I meant the mapping, it doesn't have a name or parameters

moehreag replied:
the parameter names were marked as suggestions (which I'm not entirely sure when they are put into the mappings file and when they aren't). I'm relatively certain that it's rather normal for a constructor to not have a mapped name.

IotaBread commented on mappings/net/minecraft/client/gui/widget/list/EntryListWidget.mapping:
```diff
@@ -84,6 +88,8 @@ CLASS net/minecraft/unmapped/C_nmdfcqgq net/minecraft/client/gui/widget/list/Ent
 		ARG 2 x
 		ARG 3 y
+	METHOD m_tpmgwusc isZero (I)Z
+		ARG 1 i
```
`index`?

moehreag replied:
The method body is just `return i==0` so I'm not sure what to name the parameter really..

IotaBread commented on mappings/net/minecraft/client/network/ServerInfo.mapping:
```diff
@@ -13,17 +14,22 @@ CLASS net/minecraft/unmapped/C_xotldzgg net/minecraft/client/network/ServerInfo
 	METHOD m_mgdrmoly toNbt ()Lnet/minecraft/unmapped/C_hhlwcnih;
 	METHOD m_ntjhtcjm copyWithSettingsFrom (Lnet/minecraft/unmapped/C_xotldzgg;)V
 		ARG 1 serverInfo
+	METHOD m_rvsteyeu setResult (Lnet/minecraft/unmapped/C_xotldzgg$C_odxprzma;)V
+		ARG 1 result
```
```suggestion
	METHOD m_rvsteyeu setPingResult (Lnet/minecraft/unmapped/C_xotldzgg$C_odxprzma;)V
		ARG 1 pingResult
```

moehreag committed:
address most requested changes

everything except the i/index parameter and the "empty" method, they require further discussion.

Co-Authored-By: Iota <47987888+IotaBread@users.noreply.github.com>

moehreag committed:
accept suggested parameter names for InGameHud.HeartType

moehreag committed:
correct Poisioned → Poisoned
